src/OSPSuite.Core/Domain/ModelCoreSimulationExtensions.cs

```diff
@@ -34,17 +45,18 @@ private static IReadOnlyList<IContainer> allApplicationsForMolecule(IModelCoreSi
       private static IReadOnlyList<IContainer> getApplicationsForAppliedAncestorMolecule(IEnumerable<IReactionBuilder> reactions, string moleculeName,
          IReadOnlyList<IContainer> allApplications)
       {
-         if (reactions == null)
-            return new List<IContainer>();
-
-         var reactionsList = reactions.ToList();
          var applicationsForAppliedAncestorMolecule =
             allApplications.Where(c => c.GetSingleChildByName<IMoleculeAmount>(moleculeName) != null).ToList();
 
          // If there are any applications of this molecule, use them
          if (applicationsForAppliedAncestorMolecule.Any())
             return applicationsForAppliedAncestorMolecule;
 
+         if (reactions == null)
```

Comment:
@Yuri05 I moved the check with reactions below the first if. This is probably clearer even though this is equivalent as null reaction building block in a simulation is a problem

Reply:
yep, that's more clear :)